### ISSUE TITLE
useFind causes small UI disruption when changing query params

### ISSUE THREAD

**Issue by engineertdog** [marshallswain/feathers-pinia#78]
This appears to only occur when `$sort` is set in the params sent to `useFind`, but it may occur under similar scenarios. As seen below, as the query parameters change, such as changing the `$sort` value, it causes a slight disruption in the UI before rendering the new data set.

![UI](https://i.gyazo.com/cf2ff1b184bf3c3af31ab6c26804a834.gif)

This appears to be caused by [this block](https://github.com/marshallswain/feathers-pinia/blob/v1/src/use-find.ts#L125-L130) of code in `useFind`. What ends up happening is a call to [makeUseFindItems](https://github.com/marshallswain/feathers-pinia/blob/v1/src/utils.ts#L231-L252) is made multiple times when `$sort` changes. When this occurs, the results from `makeUseFindItems` are the following:

- []
- []
- The results from the updated search query

This is ultimately the cause of the UI disruption. To ensure it's not just an implementation, here are the following code blocks currently used.

domainStore
```js
export class Domain extends BaseModel {
  _id: string;
  domain: string;
  description: string;
  createdAt: Date;
  createdBy: string;
  createdByUser?: User;
  updatedAt: Date;
  updatedBy: string;
  updatedByUser?: User;
}

const servicePath = "system/domains";
export const useDomainStore = defineStore({ servicePath, Model: Domain });
```

Vue component
```js
const domainStore = useDomainStore();
const { storeParams, pagination, loading, fetchRecords, setPaginationParams, setPagination } = useFeathersFindStore({ sortBy: "domain" });
const { data: domains, limit, currentPage, total, find } = domainStore.useFind(storeParams);

// This is called by Quasar when it is necessary to get records for the table.
const fetchDomains: QTableProps["onRequest"] = async (requestProps) => {
      setPaginationParams(requestProps);
      await fetchRecords(find, limit.value, total.value, currentPage.value);
};
```

useFeathersFindStore
```js
import { ref, reactive } from "vue";
import { QTableProps } from "quasar";
import { FindFn, BaseModel, FindClassParams } from "feathers-pinia";
import { IFeathersStore, Pagination } from "src/types";

export const useFeathersFindStore = ({
  sortBy = "_id",
  sortDirection = 1
}: IFeathersStore) => {
  const loading = ref(false);
  const pagination = ref(new Pagination(sortBy));
  const requestPagination = ref(new Pagination(sortBy));
  const storeParams = reactive<FindClassParams>({
    query: {
      $limit: 10,
      $skip: 0,
      $sort: {
        [sortBy]: sortDirection
      }
    },
    paginate: true,
    onServer: true,
    immediate: false
  });

  const setPaginationParams: QTableProps["onRequest"] = (requestProps) => {
    requestPagination.value = requestProps.pagination;
    const {
      sortBy: sort,
      descending,
      rowsPerPage,
      page
    } = requestProps.pagination;

    storeParams.query = {
      ...storeParams.query,
      $limit: rowsPerPage,
      $skip: (page - 1) * rowsPerPage,
      $sort: {
        [sort]: descending ? -1 : 1
      }
    };
  };

  const setPagination = (limit: number, total: number, currentPage: number) => {
    pagination.value = {
      sortBy: requestPagination.value.sortBy,
      descending: requestPagination.value.descending,
      rowsPerPage: limit,
      rowsNumber: total,
      page: currentPage
    };
  };

  const setLoading = (value: boolean) => {
    loading.value = value;
  };

  const fetchRecords = async (
    find: FindFn<BaseModel>,
    limit: number,
    total: number,
    currentPage: number
  ) => {
    setLoading(true);

    try {
      await find();
      setPagination(limit, total, currentPage);
    } catch (err: unknown) {
      console.log("failed", err);
    }

    setLoading(false);
  };

  return {
    storeParams,
    pagination,
    loading,
    fetchRecords,
    setPaginationParams,
    setPagination,
    setLoading
  };
};

```

**Comment by marshallswain** [comment]
@engineertdog where did we end up on this after our discussions in Discord?  Seems like we found the cause, but did you find a solution?

**Comment by engineertdog** [comment]
Our discussion had the following comment in relation to the way data is currently handled.

> Please open an issue with as much detail as you have.  The current implementation might rely too much on computed reactivity.  We may need to cache the results of the previous query in a ref and show that ref until the data is ready.  Might work.

**Comment by marshallswain** [comment]
Ah that's right.  Thanks.

**Comment by marshallswain** [comment]
I believe I have this fixed in `feathers-pinia@3.0.0-pre.13`.

**Comment by marshallswain** [comment]
This is fixed in version pre.13.
